File: 1636.sort-array-by-increasing-frequency.cpp
/*
 * @lc app=leetcode id=1636 lang=cpp
 *
 * [1636] Sort Array by Increasing Frequency
 */
#include <vector>
#include <algorithm>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<int> frequencySort(vector<int> &nums)
    {
        vector<int> freq(201, 0);
        for (auto n : nums)
        {
            ++freq[n + 100];
        }
        sort(nums.begin(), nums.end(), [&freq](int &l, int &r)
             { if(freq[l + 100] < freq[r + 100]){
                 return true;
             }else if(freq[l + 100] == freq[r + 100]){
                 return l>r;
             }else{
                 return false;
             } });
        return nums;
    }
};
// @lc code=end

File: py/1636.sort-array-by-increasing-frequency.py
#
# @lc app=leetcode id=1636 lang=python3
#
# [1636] Sort Array by Increasing Frequency
#

# @lc code=start
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        freq = [0]*201
        for n in nums:
            freq[n+100] += 1
        nums.sort(key=lambda x: (freq[x+100] << 10)-x)
        return nums
# @lc code=end
